sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/metadata/ApiClientConfigurationTest.java

```diff
@@ -0,0 +1,98 @@
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+/** Test that http client configuration is correct. */
+@SpringBootTest(
+    classes = {AuthTestApp.class, MetadataTestApp.class},
+    webEnvironment = RANDOM_PORT,
+    // verify authentication in this test, not authorization
+    properties = {
+      "opa.disable=true",
+      "metadata.other.baseUrl=http://localhost:${wiremock.server.port}/api",
+      "metadata.otherAuthenticated.baseUrl=http://localhost:${wiremock.server.port}/api",
+    })
+@ContextConfiguration(initializers = EnableSdaAuthMockInitializer.class)
+@SetSystemProperty(key = "METADATA_FIELDS", value = "tenant-id")
+@AutoConfigureMockMvc
+class ApiClientConfigurationTest {
+  @Autowired private MockMvc mockMvc;
+
+  @MockBean private MetadataOtherServiceClient metadataOtherServiceClient;
+
+  @MockBean private MetadataOtherServiceAuthenticatedClient metadataOtherServiceAuthenticatedClient;
+
+  @Test
+  void submitMetadataContextInPlatformClient() throws Exception {
+    DetachedMetadataContext metadataContext = new DetachedMetadataContext();
+    metadataContext.put("tenant-id", List.of("t-1"));
+    MetadataContext.createContext(metadataContext);
+
+    var httpHeaders = new HttpHeaders();
+    httpHeaders.put("tenant-id", List.of("t-1"));
+
+    when(metadataOtherServiceAuthenticatedClient.getSomething(httpHeaders))
+        .thenReturn(Map.of("hello", "world"));
+
+    mockMvc
+        .perform(
+            MockMvcRequestBuilders.request(HttpMethod.GET, "/metadataAuthProxy")
+                .headers(httpHeaders)
+                .accept(APPLICATION_JSON))
+        .andExpect(MockMvcResultMatchers.status().isOk());
+    verify(metadataOtherServiceAuthenticatedClient).getSomething(httpHeaders);
+
+    MetadataContext.createContext(new DetachedMetadataContext());
+  }
+
+  @Test
+  void dontSubmitMetadataContextInExternalClient() throws Exception {
+    DetachedMetadataContext metadataContext = new DetachedMetadataContext();
+    metadataContext.put("tenant-id", List.of("t-1"));
+    MetadataContext.createContext(metadataContext);
+
+    var httpHeaders = new HttpHeaders();
+    httpHeaders.put("tenant-id", List.of("t-1"));
+
+    when(metadataOtherServiceAuthenticatedClient.getSomething(httpHeaders))
+        .thenReturn(Map.of("hello", "world"));
+
+    mockMvc
+        .perform(
+            MockMvcRequestBuilders.request(HttpMethod.GET, "/metadataProxy")
+                .headers(httpHeaders)
+                .accept(APPLICATION_JSON))
+        .andExpect(MockMvcResultMatchers.status().isOk());
+
+    verify(metadataOtherServiceClient).getSomething(httpHeaders);
```

Comment:
Don't we expect blank headers to be sent to other/external services? `httpHeaders` here is the same as in the test above.